File: Active/Go/solutions/p002.go
// Project Euler
// Problem 2: The sum of even Fibonacci sequence below 4_000_000

package go_euler

func Problem_002(limit int) int {
	sum := 0
	// the first 2 number of the Fibonacci
	var x, y int = 1, 2
	// temporarry variable to store intermediate value
	var temp int = 0

	for y < limit {
		if y%2 == 0 {
			sum += y
		}

		temp = x
		x = y
		y = temp + x
	}

	return sum
}

File: Inactive/Java/P002.java
// Project Euler
// Problem 2: The sum of even Fibonacci number under 4000000

public class P002 implements Solution<Integer> {
    public static void main(String[] args) {
        int limit = 4000_000;
        System.out.println(new P002().compute(limit));
    }

    public String compute(Integer limit) {
        int x = 0;  // the first  number of Fibonacci
        int y = 1;  // the second number of Fibonacci
        int temp = 0;   // temporary variable to store value
        int sum = 0;

        while (y < limit) {
            if (y % 2 == 0) sum += y;

            temp = x;
            x = y;
            y = x + temp;
        }

        return Integer.toString(sum);
    }
}
